libbeat/autodiscover/autodiscover.go

```diff
@@ -65,10 +63,7 @@ type Autodiscover struct {
 	meta            *meta.Map
 	listener        bus.Listener
 	logger          *logp.Logger
-
-	// workDone is a channel used for testing purpouses, to know when the worker has
-	// done some work.
-	workDone chan struct{}
+	debouncePeriod  time.Duration
```

Comment:
I'm probably overlooking something but I'm not following why this field is needed on the struct.  In the previous implementation, the `worker()` method was directly using the const defined for the debounce period.  It looks like now this struct field is being initialized with the same const and then the `worker()` method is referencing the field instead of the const.  What would happen if this field were removed and the `worker()` method directly references the const instead?

Reply:
I moved it to the struct so I could change it for testing. In the end I ended up not modifying it and just going with the default.

I can change it back to just being the constant if you prefer.

Reply:
Yeah, that was my first guess too - that you didn't want it as a constant for testing reasons.  

Anyway, I think either we use it in testing or we move it back to a const.  Otherwise, it's a bit confusing the way it is, IMO.

Reply:
Thanks @ycombinator, I ended up using it and speeding up some of the tests by almost 4x!